### PR TITLE
Add Switcher, Button, TextField, Label modules.

### PR DESCRIPTION
**Pull request by timoxley**
Adds Switcher, Button, TextField, Label modules.

For testing Switcher & Label:
* Connect a Switcher to a Label.
* Toggle Switch, try undo/redo.
* Start canvas, note Label shows a value.
* Toggle Switch, note Label value updates.
* Stop canvas. Note switch reverts to value used at canvas start.
* Open same canvas in another tab.
* Start canvas.
* Toggle switch.
* Note switch & label value is replicated to other tab.

For testing Button & TextField:

* Add a Button.
* Change the button value param. Note button value changes.
* Add a TextField.
* Edit TextField content.
* Content is saved on blur. Try undo/redo.
* Connect TextField's out to Button's name.
* Start canvas
* Note actual button value changes.
* Edit TextField content.
* Press send button.
* Note actual button value changes with new textfield content.
* Stop Canvas.
* Note value changes are reverted.

Also note that Textfield & Comment content changes won't replicate across tabs at runtime, the changed values currently don't get sent over the subscription.

Also: 
* Improves API for doing subscriptions & sending data.
* Fixes undo/redo for comment module. Need to create wipe DOM nodes for undo-managed text inputs on blur otherwise in-browser undo gets confused with the app's undo.
* Updates Button name value from button's `buttonName` param.
* Adjusts default module sizes.
* Makes Table resizable.
* Adds a bunch of explanatory comments in the code.

Closes PLATFORM-255 (switcher), PLATFORM-256 (textfield), PLATFORM-247 (label), PLATFORM-331 (remove getState)

### CODE REVIEW

File: app/src/editor/shared/components/Client.jsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        this.teardown()
+    }
+
+    setup() {
+        const { authKey } = this.props
+        if (!authKey || this.state.client) { return }
+        this.setState({
+            client: new StreamrClient({
+                url: process.env.STREAMR_WS_URL,
+                authKey,
```
Comment (juhah) [inline app/src/editor/shared/components/Client.jsx]: We are using `streams-client` v.0.12.2, this should be `apiKey` (https://github.com/streamr-dev/streamr-client-javascript/tree/v0.12.2). But if it works, probably best to leave it and update to client 1.0.0 at some point.

File: app/src/editor/canvas/state.js [hunk 2]
```diff
@@ -444,17 +453,19 @@ export function removeModule(canvas, moduleHash) {
...
 export function addModule(canvas, moduleData) {
-    const canvasModule = { ...moduleData }
-    canvasModule.hash = Date.now()
-    canvasModule.layout = {
-        position: {
-            top: 0,
-            left: 0,
+    const canvasModule = {
+        ...moduleData,
+        hash: Date.now(), // TODO: better ids
```
[the remark this answers is not in the document]
Comment (timoxley) [reply app/src/editor/canvas/state.js]: @juhah `hash` is the real "id" of the module, relative to the canvas anyway. Client is currently in charge of setting these as modules are added to the canvas object before the server actually sees them.

File: app/src/editor/shared/components/Client.jsx (hunk 1 above)
Comment (timoxley) [reply app/src/editor/shared/components/Client.jsx]: Yeah it seems to work.

File: app/src/editor/shared/components/ModuleLoader.jsx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+import React from 'react'
+
+import { ClientContext } from './Client'
+
+/**
+ * Supplies default implementation of loadModule to ModuleLoader.
+ * Separated from ModuleLoader so ModuleLoader core logic can be tested without magic mocks.
+ */
+
+export default class ModuleLoaderContainer extends React.PureComponent {
```
Comment (tumppi) [inline app/src/editor/shared/components/ModuleLoader.jsx]: 👍

**Pull request merged**